dev-docs/yaml-design-guidance.md

```diff
@@ -127,6 +128,38 @@ struct Export {
 ```
 Take a look at `env_defaults` in `expansion.rs` to see how env variables should be defaulted.
 
+### Use `#[serde(default)]` on struct instead of fields when possible
+If all the fields of your struct have their default value then use the `#[serde(default)]` on the struct instead of all fields. If you have specific default values for field, you have to create your own `Default` impl.
```

Comment:
maybe add some context on why it should be done this way (ax explained in https://github.com/apollographql/router/issues/2424)

Reply:
Fixed